app/controllers/api/v2/table_preferences_controller.rb

```diff
@@ -7,20 +7,24 @@ class TablePreferencesController < V2::BaseController
       before_action :find_resource, :only => [:destroy, :update, :show]
 
       api :GET, "/users/:user_id/table_preferences", N_("List of table preferences for a user")
+      param :user_id, String, :desc => N_("ID of the user"), :required => true
+      param_group :search_and_pagination, ::Api::V2::BaseController
       def index
         @table_preferences = @user.table_preferences
       end
 
       api :GET, "/users/:user_id/table_preferences/:name", N_("Table preference details of a given table")
-      param :name, String, :required => true
+      param :user_id, String, :desc => N_("ID of the user"), :required => true
+      param :name, String, :desc => N_("Name of the table"), :required => true
       def show
         if @table_preference.blank?
           @table_preference = TablePreference.new(:user => @user, :name => params[:name])
         end
       end
 
       def_param_group :table_preference do
-        param :table_preferences, Hash, :required => true do
+        param :user_id, String, :desc => N_('ID of the user'), :required => true
+        param :table_preference, Hash, :required => true do
           param :name, String, :required => true, :desc => N_("Name of the table")
```

Comment:
Since this param is in the group, it will be applied for both create and update. I think we need to extract this and apply for create only.

Reply:
I think the issue was that it had nested parameter so instead of that I created group without nested parameter

Reply:
Well, but still https://github.com/theforeman/foreman/pull/9305#discussion_r923268927